planner/cascades/implementation_rules.go

```diff
@@ -436,3 +442,49 @@ func (r *ImplUnionAll) OnImplement(expr *memo.GroupExpr, reqProp *property.Physi
 	physicalUnion.SetSchema(expr.Group.Prop.Schema)
 	return impl.NewUnionAllImpl(physicalUnion), nil
 }
+
+// ImplApply implements LogicalApply to PhysicalApply
+type ImplApply struct {
+}
+
+// Match implements ImplementationRule Match interface.
+func (r *ImplApply) Match(expr *memo.GroupExpr, prop *property.PhysicalProperty) (matched bool) {
+	return true
```

Comment:
Shall we check the property here? The old implementation uses:
```
if !prop.AllColsFromSchema(la.children[0].Schema()) { // for convenient, we don't pass through any prop
		return nil
	}
```

Reply:
We can push the prop down and let its children check. But this check is a better way to reduce calculation. I'll change it.